### DISCUSSION TITLE
SOLVED - Suddenly getting invalid TOTP message - can't login

### DISCUSSION THREAD

**Discussion by burntoc** [dani-garcia/vaultwarden#3184] [Q&A] [answered]
After having used the bitwarden.rs and now vaultwarden container on Unraid for a couple of years, I am suddenly getting an invalid TOTP message and the timestamp shows the UTC time, which is 8 hours ahead of myy timezone. I don't know why this is happening or how to fix it.  This is what I've tried so far:
Check server time via settings and via "date" command in console - shows correct local time
Check container time via "date" command in console - shows correct local time
Restart container - issue persists
Restart client - issue persists
set TZ variable (never needed before) - issue persists
Tried other clients (that have also worked for a long time) - same issue

UPDATE - Unfortunately, I must have cleared my cookies so now I can't log in to BW at all. (though the extensions that have already worked continue to do so with their cached biometric auth).  

I could definitely use help understantding how to fix this issue, which would be best.  Worst case, since I have physical access to my host server and if I can't find my recovery phrase, is there something I can edit in the appdata folder that will allow me to disable TOTP and just login with the master accound and password so I can then disable 2FA until we figure out what the heck is suddently going on?  Thanks.

**Comment by BlackDex** [comment]
Could you post the support string which you can generate at the `/admin/diagnostics` page?

Also, done log lines from the time you tried to login using TOTP could be useful.

**Comment by burntoc** [comment]
I remembered I could  get to the admin panel, and maybe the error is client drift?  If so, I don't understand why suddently multiple clients are having this failure.

![image](https://user-images.githubusercontent.com/11756265/215251833-50ddf377-d6c3-4217-9713-a74b065a99c9.png)

**Reply by BlackDex** [reply]
It could also be server drift. Check the NTP sync on the server. It seems to be about a minute. Which is to much.

**Reply by burntoc** [reply]
The host server NTP and client seem to be working and correct.  Not sure if I need to console into the container to enable it there, too?  If so that seems strange as I've never had to do it before.  

Not sure what the preferred way is to get the support string in, so pasting as a text file here.

[vaultwarden_support_string.txt](https://github.com/dani-garcia/vaultwarden/files/10526165/vaultwarden_support_string.txt)

**Reply by BlackDex** [reply]
Posting as text would have sufficed.

It really is at drift somewhere. And i doubt it would be all your clients.
So that leaves the server. I suggest to do s manual check then, like printing the time on both the server and a client at the exact same time. But then again, that is what the diagnostics page also does.

But that could maybe help you determine where the issues is.

**Reply by BlackDex** [reply]
There probably are lots of mobile apps btw which show you the atomic clock. Try those and match it with the server.

**Reply by burntoc** [reply]
Headed out to work and will check tonight when I get home.  On a related note, in the event it takes me a day or two of failed tests to get this resolved, can I modify something in the config in the appdata folder that will disable TOTP until I can get a green diagnostic result?

**Reply by BlackDex** [reply]
You need to login into the admin, and then you can only fully remove all MFA config. That will grant you the ability to login without MFA at least.

**Reply by burntoc** [reply]
> You need to login into the admin, and then you can only fully remove all MFA config. That will grant you the ability to login without MFA at least.

The admin page via the token?  Because that I can do, but I can't tell what settings to disable.

**Reply by BlackDex** [reply]
It's not a setting. You need to go to the users page, and disable/remove 2fa/mfa for the specific user you want to be able to login without 2fa/mfa.

**Reply by burntoc** [reply]
Thank you for this.  That worked great for helping me out in the interim.

**Comment by burntoc** [comment]
Hmm, so I'm hearing that somehow the Unraid NTP is off - by almost a minute
I take it since my phone showing the same page (the above came from my
laptop) show about the same delta.

Gueas I'll see about disabling NTP for a bit, or rebooting the server
altogether. It's about midnight here and work tomorrow until late day so it
may be late tomorrow before my next update. Thanks for your message.

**Comment by burntoc** [comment, marked as answer]
By way of an update, it turns out Unraid wasn't able to reach the NTP servers though the interface gave no indication as such.  Once I tracked down the Unraid and firewall behavior that was causing that issue, I was able to update the Unraid time via NTP and now the diagnostics are showing green again, so I'm sure TOTP will work - I just have to set it up for that user again.  Thanks for all the very timely help, folks!

**Reply by jimmyo-q** [reply]
Could you provide any more detail on how you were able to correct this issue?

I seem to be having the exact same problem

![chrome_58J3bSJBqR](https://github.com/dani-garcia/vaultwarden/assets/145996033/a83af5b2-4517-4f49-a35b-3c487f922f66)

**Reply by burntoc** [reply]
Looks like your server is showing a time that is about 42 seconds behind the actual NTP time because it cannot reach the NTP servers using the NTP protocol, UDP port 123.  That time difference is resulting in your client and server tokens not matching.

Whatever firewall/router you have is likely blocking it, unless you're running your vaultwarden server on a PC or workstation that has an on-host firewall blocking it - which is highly unlikely.  

As in my case, even if you aren't seeing drops you may want  to ensure you have an explicit allow rule for that specific traffic and that you're logging it - at least temporarily.  When you get a green in Server NTP and Browser NTP I imagine you'll be good to go.
